pkg/stub/handler.go

```diff
@@ -554,17 +574,10 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		h.mutex.Lock()
 		defer h.mutex.Unlock()
 
-		if h.samplesResource != nil {
-			currVersion, _ := strconv.Atoi(h.samplesResource.ResourceVersion)
-			newVersion, _ := strconv.Atoi(srcfg.ResourceVersion)
-			if newVersion <= currVersion {
-				return nil
-			}
-
-			err := h.SpecValidation(srcfg)
-			if err != nil {
-				return err
-			}
+		err = h.SpecValidation(srcfg)
```

Comment:
the operator-sdk Update() function mutates the object you pass in, updating it to the result of the update call.  This meant that when we would do something like:

Update(h.samplesResource), the cached value would get updated w/ the new resource version.

That meant when we then got back to this point based on the resulting update event, we'd ignore the update event.

So when we went through a flow of:

1) samples resource created
2) event handler see create event and instantiates samples and updates condition to samplesexist
3) event handler sees update event, checks resource version and ignores it (never calling SpecValidation and thus never setting ConfigValid to true)

i'm not entirely sure whether i broke something to cause that, or it was the state i was in, but regardless I think we should just do this every time we see the object.  I know i previously argued to avoid that, but i'm now leaning towards correctness/guarantees over performance, at least until proven otherwise.

Reply:
i'm still not convinced we should need to be capturing/using the last processed resource version at all, but again we can revisit that as we move forward.

Reply:
ok sure thing ... my concern was that when the relist came in, if we did not ignore it, we would update the images streams again, albeit with the same spec, and I was afraid it would still be firing image change triggers everywhere

Reply:
> my concern was that when the relist came in, if we did not ignore it, we would update the images streams again, albeit with the same spec, and I was afraid it would still be firing image change triggers everywhere

that should not be a problem.  we'd only fire the triggers if the import found a new sha.

now it's possible the "replace" action will trigger a re-import which could update a sha if there was a new image available (kinda like a scheduled import would) which would be undesirable behavior, but i think we need to think about that holistically (how the operator knows whether it needs to apply a new imagstream update or not.  After all, if we update the operator with new imagestreamtags for an existing imagestream, we absolutely *do* want to apply that change, even if nothing about SamplesResource itself has changed).  Again, item for future discussion.